### PR TITLE
Ternary if and remove redundant parenthesizes

### PR DESCRIPTION
**Pull request by cclauss**
tuples are more compact and unmodifiable than lists.

### CODE REVIEW

File: app/server.py [hunk 1]
```diff
@@ -25,12 +25,8 @@ def index():
...
 def bad_request(err):
     message = {'Error': err[1], 'Status Code': err[0]}
-    if(err[2] == 'xml'):
-        response = dicttoxml(message)
-    else:
-        response = json.dumps(message)
-    resp = make_response(response, err[0])
-    return resp
+    response = dicttoxml(message) if err[2] == 'xml' else json.dumps(message)
+    return make_response(response, err[0])
```
Comment (bhaveshAn) [inline app/server.py]: Does it make any difference.

File: app/server.py [hunk 2]
```diff
@@ -39,11 +35,11 @@ def search(search_engine):
         num = request.args.get('num') or 10
         count = int(num)
         qformat = request.args.get('format') or 'json'
-        if qformat not in ['json', 'xml']:
+        if qformat not in ('json', 'xml'):
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py [hunk 3]
```diff
@@ -39,11 +35,11 @@ def search(search_engine):
...
             abort(400, 'Not Found - undefined format')
 
         engine = search_engine
-        if engine not in ['google', 'bing', 'duckduckgo', 'yahoo', 'ask']:
+        if engine not in ('google', 'bing', 'duckduckgo', 'yahoo', 'ask'):
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py [hunk 4]
```diff
@@ -57,17 +53,17 @@ def search(search_engine):
             err = [404, 'No response', qformat]
             return bad_request(err)
 
-        if((db['queries'].find({query: query}).limit(1)) is False):
+        if db['queries'].find({query: query}).limit(1) is False:
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py [hunk 5]
```diff
@@ -57,17 +53,17 @@ def search(search_engine):
...
             db['queries'].insert(
                 {"query": query, "engine": engine, "qformat": qformat})
 
         for line in result:
             line['link'] = line['link'].encode('utf-8')
             line['title'] = line['title'].encode('utf-8')
-            if(engine == 'b'):
+            if engine == 'b':
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py [hunk 6]
```diff
@@ -57,17 +53,17 @@ def search(search_engine):
...
                 line['desc'] = line['desc'].encode('utf-8')
 
-        if(qformat == 'json'):
+        if qformat == 'json':
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py [hunk 7]
```diff
@@ -94,7 +90,7 @@ def set_header(r):
     app.debug = True
     app.run(
         host='0.0.0.0',
-        port=(int)(
+        port=int(
```
Comment (bhaveshAn) [inline app/server.py]: Same

File: app/server.py (hunk 1 above)
Comment (cclauss) [reply app/server.py]: Perhaps run dis.dis() on the code and see.

File: app/server.py (hunk 2 above)
Comment (cclauss) [reply app/server.py]: As it says in the comments above, tuples are more compact and unmodifiable than lists.

File: app/server.py (hunk 3 above)
Comment (cclauss) [reply app/server.py]: As it says in the comments above, tuples are more compact and unmodifiable than lists.

File: app/server.py (hunk 5 above)
Comment (cclauss) [reply app/server.py]: It merely more Pythonic.

File: app/server.py (hunk 4 above)
Comment (cclauss) [reply app/server.py]: It merely more Pythonic.  It is not JavaScript or Lisp.

File: app/server.py (hunk 7 above)
Comment (cclauss) [reply app/server.py]: It merely more Pythonic.

File: app/server.py (hunk 6 above)
Comment (cclauss) [reply app/server.py]: It merely more Pythonic.

**Comment by vaibhavsingh97** [comment]
@bhaveshAn I see this PR is perfect. I mean the changes made by @cclauss is gud and it's more pythonic. Also, official documentation also follow these. please check here https://docs.python.org/3/tutorial/controlflow.html

**Pull request merged**